src/frametree.cpp

```diff
@@ -820,12 +832,9 @@ int FrameTree::splitCommand(Input input, Output output)
     SplitAlign align_auto = (lw > lh) ? SplitAlign::horizontal : SplitAlign::vertical;
     SplitAlign align_explode = SplitAlign::vertical;
     auto availableModes = SplitMode::modes(align_explode, align_auto);
-    SplitMode m;
-    for (auto &it : availableModes) {
-        if (it.name[0] == splitType[0]) {
-            m = it;
-        }
-    }
+    SplitMode m = *std::find_if(
+            availableModes.begin(), availableModes.end(),
+            [=](const SplitMode &x){ return x.name[0] == splitType[0]; });
```

Comment:
What happens here if `std::find_if` returns `availableModes.end()`?

Reply:
Ooh, good catch! I've added a test for this in #825, which would have caught this blunder.

Reply:
(Oh and, in case your question wasn't rhetorical: it actually crashes...)

Reply:
Ah, I strongly suspected that but wasn't sure :-)